cli/azd/pkg/commands/pipeline/azdo_provider.go

```diff
@@ -429,6 +428,44 @@ func (p *AzdoScmProvider) gitRepoDetails(ctx context.Context, remoteUrl string)
 		repoDetails.remoteUrl = remoteUrl
 	}
 
+	if repoDetails.projectId == "" || repoDetails.repoId == "" {
+		// Removing environment or creating a new one would remove any memory fro project
+		// and repo.  In that case, it needs to be calculated from the remote url
+		azdoSlug, err := parseAzDoRemote(remoteUrl)
+		if err != nil {
+			return nil, fmt.Errorf("parsing Azure DevOps remote url: %s: %w", remoteUrl, err)
+		}
+		// azdoSlug => Org/Project/_git/repoName
```

Comment:
Not sure if this is worth fixing, but there exists old but working Azure DevOps URL in the form of <Project>.visualstudio.com

Example:
https://devdiv.visualstudio.com/DevDiv/_git/AddressSanitizer

Reply:
I would just wait to hear/see if someone creates a bug for it

Reply:
I think we can just include this as unsupported for now.. It's long been deprecated but I'm surprised these links are still working